WalletWasabi/Legal/LegalDocuments.cs

```diff
@@ -10,63 +10,84 @@ namespace WalletWasabi.Legal
 {
 	public class LegalDocuments
 	{
-		public const string EmbeddedFileName = "LegalDocuments.txt";
-		public const string LegalFolderName = "Legal";
-		public const string AssetsFoldername = "Assets";
-		public static readonly string EmbeddedFilePath = Path.Combine(EnvironmentHelpers.GetFullBaseDirectory(), LegalFolderName, AssetsFoldername, EmbeddedFileName);
+		public static readonly string EmbeddedFilePath = Path.Combine(EnvironmentHelpers.GetFullBaseDirectory(), "Legal", "Assets", "LegalDocuments.txt");
 
-		public LegalDocuments(string filePath)
+		public LegalDocuments(Version version, string content)
 		{
-			FilePath = Guard.NotNullOrEmptyOrWhitespace(nameof(filePath), filePath, trim: true);
-			var verString = Path.GetFileNameWithoutExtension(FilePath);
-			Version = Version.Parse(verString);
+			Version = version;
+			Content = content;
 		}
 
-		public string FilePath { get; }
 		public Version Version { get; }
+		public string Content { get; }
 
-		public static LegalDocuments TryLoadAgreed(string dataDir)
+		public static async Task<LegalDocuments?> TryLoadAgreedAsync(string folderPath)
 		{
-			var legalFolderPath = Path.Combine(dataDir, LegalFolderName);
-			IoHelpers.EnsureDirectoryExists(legalFolderPath);
-			var legalDocCandidates = FindCandidates(legalFolderPath);
+			if (!Directory.Exists(folderPath))
+			{
+				return null;
+			}
+
+			var legalDocCandidates = FindCandidates(folderPath);
 			var legalDocCandidateCount = legalDocCandidates.Count();
 
 			if (legalDocCandidateCount == 1)
 			{
 				var filePath = legalDocCandidates.Single();
-				return new LegalDocuments(filePath);
+
+				var content = await File.ReadAllTextAsync(filePath).ConfigureAwait(false);
+				var verString = Path.GetFileNameWithoutExtension(filePath);
```

Comment:
One issue with `var`s is that when somebody reads the code, she needs to:

* guess type (obviously)
* guess whether type is nullable or not. That is harder and not that obvious.

So when I examine in my MSVS return type of `Path.GetFileNameWithoutExtension`, it is `string?` but when you look here https://source.dot.net/#System.Private.CoreLib/Path.cs,e36b3edc1f3e2c8d it is actually `string` in this case.

So writing code is faster with `var` and reading code is harder for everyone. That's why I don't like vars mostly.

Reply:
ACK